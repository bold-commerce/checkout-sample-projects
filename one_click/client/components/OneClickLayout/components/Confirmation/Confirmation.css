.ConfirmationPage {
    display: grid;
    gap: 16px;
}

.ConfirmationPage h1 {
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    margin: 16px, 0px;
    color: rgba(0, 0, 0, 0.9);
}

.ConfirmationPage h2 {
    font-size: 18px;
    font-weight: normal;
    line-height: 20px;
    margin-bottom: 16px;
}

.ConfirmationPage h3 {
    margin-top: 16px;
    font-weight: 700;
    font-size: 12px;
    line-height: 16px;
}

.ConfirmationPage p {
    margin-top: 8px;
    font-weight: normal;
    font-size: 12px;
    line-height: 16px;
    color: rgba(0, 0, 0, .65);
}

.confirmation-order,
.confirmation-information {
    color: rgba(0, 0, 0, 0.9);
    border: 1px solid #d9d9d9;
    padding: 16px;
    margin: 16px, 0px;
}

.confirmation-order-accepted {
    font-size: 14px;
    line-height: 20px;
    font-weight: normal;
    color: rgba(0, 0, 0, .65);
}

.confirmation-continue {
    width: 100%;
    margin: 16px 0px;
    color: white;
    background-color: #0a75c2;
    display: revert;
}