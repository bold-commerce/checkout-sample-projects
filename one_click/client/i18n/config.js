import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;