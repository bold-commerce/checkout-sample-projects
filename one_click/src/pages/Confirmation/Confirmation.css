.ConfirmationPage div {
    margin: 16px 0;
}

.ConfirmationPage .ThankYou {
    font-size: 20px;
    line-height: 20px;
    font-weight: normal;
    color: rgba(0, 0, 0, 0.9);
}

.ConfirmationPage h2 {
    font-size: 18px;
    font-weight: normal;
    line-height: 20px;
    margin-bottom: 16px;
}

.ConfirmationPage h3 {
    margin-top: 16px;
    font-weight: 700;
    font-size: 12px;
    line-height: 16px;
}

.ConfirmationPage p {
    margin: 4px 0px;
    font-weight: normal;
    font-size: 12px;
    line-height: 16px;
    color: rgba(0, 0, 0, .65);
}

.ConfirmationPage p:first-child {
    margin-top: 0px;
}

.confirmation-order,
.confirmation-information {
    color: rgba(0, 0, 0, 0.9);
    border: 1px solid #d9d9d9;
    border-radius: 4px;
    padding: 16px;
}

.confirmation-order-accepted {
    font-size: 14px;
    line-height: 20px;
    font-weight: normal;
    color: rgba(0, 0, 0, .65);
}

button.confirmation-continue-button {
    width: 100%; 
    padding: 8px;
    border-radius: 4px;
    box-shadow: 0px 1px 2px 0px #0000001A;
    border: 1px solid #000000A6;
}

.confirmation-help p {
    font-size: 14px;
}

.confirmation-help p a,
.confirmation-help p a:hover{
    color: #0a75c2
}
