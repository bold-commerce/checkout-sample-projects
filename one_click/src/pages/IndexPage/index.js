export { default as IndexPage } from "./IndexPage";
export { default as IndexPageGuest } from "./IndexPageGuest";