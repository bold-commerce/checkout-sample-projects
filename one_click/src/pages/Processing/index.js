export { default as ProcessingOrder } from './Processing'