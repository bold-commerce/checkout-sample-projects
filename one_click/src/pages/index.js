export * from './Confirmation';
export * from './IndexPage';
export * from './Processing';
export * from './Inventory';