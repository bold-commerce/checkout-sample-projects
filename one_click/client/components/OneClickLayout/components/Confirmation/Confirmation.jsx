import React, { useEffect, useContext } from 'react';
import { useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { Link } from 'react-router-dom';
import { useAnalytics } from '../../../../hooks';
import { AppContext } from '../../context/AppContext';
import { Header } from '../Header';
import './Confirmation.css'

const Confirmation = ({}, ref) => {
    const { state } = useCheckoutStore();
    const { addresses, customer, payments, shipping} = state.applicationState;
    const trackEvent = useAnalytics();
    const { websiteName } = useContext(AppContext);

    useEffect(() => {
      trackEvent('thank_you');
    }, []);

    return(
        <div ref={ref} className='Sidebar ConfirmationPage'>
            <Header title={websiteName} />
            <div className="ThankYou">Thank you{ customer.first_name && ', ' + customer.first_name }!</div>
            <div className='confirmation-order'>
                <h2>Your order is confirmed</h2>
                <div className='confirmation-order-accepted'>We've accepted your order, and we're getting it ready. A confirmation email has been sent to your email address.</div>
            </div>
            <div className='confirmation-information'>
                <h2>Customer information</h2>
                <h3>Shipping addrees</h3>
                <p>{addresses.shipping.first_name} {addresses.shipping.last_name}</p>
                <p>{addresses.shipping.address_line_1}</p>
                <p>{addresses.shipping.address_line_2}</p>
                <p>{addresses.shipping.city} {addresses.shipping.province_code} {addresses.shipping.postal_code}</p>
                <p>{addresses.shipping.country}</p>

                <h3>Billing address</h3>
                <p>{addresses.billing.first_name} {addresses.billing.last_name}</p>
                <p>{addresses.billing.address_line_1}</p>
                <p>{addresses.billing.address_line_2}</p>
                <p>{addresses.billing.city} {addresses.billing.province_code} {addresses.billing.postal_code}</p>
                <p>{addresses.billing.country}</p>

                <h3>Shipping method</h3>
                <p>{shipping.selected_shipping.description}</p>

                <h3>Payment method</h3>
                {
                    payments &&
                    payments.map( (payment) => {
                        return <p key={payment.id} >{payment.brand}</p>
                    })
                }
            </div>
            <div className='confirmation-help'>
                <p>Need help? <a href='#'>Contact us</a></p> { /* TODO: get correct contact us link */}
            </div>
            <div className="confirmation-continue">
                <Link className='confirmation-continue-button stx-button' to='/' >
                    Continue shopping
                </Link>
            </div>
        </div>
    );
};

const ConfirmationForwardedRef = React.forwardRef(Confirmation);

export default ConfirmationForwardedRef;