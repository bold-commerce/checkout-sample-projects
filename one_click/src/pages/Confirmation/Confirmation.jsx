import React, { useEffect, useContext, useCallback } from 'react';
import { useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { useAnalytics } from '../../hooks';
import { AppContext } from '../../context/AppContext';
import { Header } from '../../components/Header';
import './Confirmation.css'
import { useTranslation } from 'react-i18next';
import { Button } from '@boldcommerce/stacks-ui';

const Confirmation = ({}, ref) => {
    const { state } = useCheckoutStore();
    const { addresses, payments, shipping} = state.applicationState;
    const trackEvent = useAnalytics();
    const { websiteName } = useContext(AppContext);
    const { t } = useTranslation();

    useEffect(() => {
      trackEvent('thank_you');
    }, []);

    const continueShopping = useCallback(() => {
        const event = new CustomEvent("oneClick:close");
        window.dispatchEvent(event);
    }, []);

    return(
        <div ref={ref} className='Sidebar ConfirmationPage'>
            <Header title={websiteName} />
            <div className="ThankYou">{t('confirmation.thank_you')}</div>
            <div className='confirmation-order'>
                <h2>{t('confirmation.order_confirmed')}</h2>
                <div className='confirmation-order-accepted'>{t('confirmation.order_accepted')}</div>
            </div>
            <div className='confirmation-information'>
                <h2>{t('customer.info')}</h2>
                <h3>{t('shipping.address')}</h3>
                <p>{addresses.shipping.first_name} {addresses.shipping.last_name}</p>
                <p>{addresses.shipping.address_line_1}</p>
                <p>{addresses.shipping.address_line_2}</p>
                <p>{addresses.shipping.city} {addresses.shipping.province_code} {addresses.shipping.postal_code}</p>
                <p>{addresses.shipping.country}</p>

                <h3>{t('billing.address')}</h3>
                <p>{addresses.billing.first_name} {addresses.billing.last_name}</p>
                <p>{addresses.billing.address_line_1}</p>
                <p>{addresses.billing.address_line_2}</p>
                <p>{addresses.billing.city} {addresses.billing.province_code} {addresses.billing.postal_code}</p>
                <p>{addresses.billing.country}</p>

                <h3>{t('shipping.method')}</h3>
                <p>{shipping.selected_shipping.description}</p>

                <h3>{t('payment.method')}</h3>
                {
                    payments &&
                    payments.map( (payment) => {
                        return <p key={payment.id} >{payment.brand}</p>
                    })
                }
            </div>
            <div className='confirmation-help'>
                <p>{t('confirmation.need_help')}<a href='#'>{t('confirmation.contact_us')}</a></p> { /* TODO: get correct contact us link */}
            </div>
            <div className="confirmation-continue">
                <Button className='confirmation-continue-button' onClick={continueShopping} > 
                    {t('confirmation.continue_shopping')}
                </Button>
            </div>
        </div>
    );
};

const ConfirmationForwardedRef = React.forwardRef(Confirmation);

export default ConfirmationForwardedRef;