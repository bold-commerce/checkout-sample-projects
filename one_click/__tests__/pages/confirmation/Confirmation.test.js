import React from 'react';
import { Confirmation } from '../../../src/pages/Confirmation';
import { testApplicationState as MOCKtestApplicationState } from '../../utils/applicationStateHelper';
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import '../../../src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useCheckoutStore: () => ({
        state: { applicationState: MOCKtestApplicationState }
    })
})).mock('React', () => ({
    ...jest.requireActual('React'),
    useContext: () => ({
        websiteName: 'app.test'
    })
}));

describe('Confirmation', () => {
    test('renders Confirmation component', () => {
        const { asFragment } = render(
            <Confirmation />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    });
});