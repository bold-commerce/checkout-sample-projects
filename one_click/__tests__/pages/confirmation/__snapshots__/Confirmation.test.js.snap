// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation renders Confirmation component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar ConfirmationPage"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        app.test
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="ThankYou"
    >
      confirmation.thank_you
    </div>
    <div
      class="confirmation-order"
    >
      <h2>
        confirmation.order_confirmed
      </h2>
      <div
        class="confirmation-order-accepted"
      >
        confirmation.order_accepted
      </div>
    </div>
    <div
      class="confirmation-information"
    >
      <h2>
        customer.info
      </h2>
      <h3>
        shipping.address
      </h3>
      <p>
        John Doe
      </p>
      <p>
        123 Fake Street
      </p>
      <p />
      <p>
        Winnipeg MB R3T 1B8
      </p>
      <p>
        Canada
      </p>
      <h3>
        billing.address
      </h3>
      <p>
        John Doe
      </p>
      <p>
        123 Fake Street
      </p>
      <p />
      <p>
        Winnipeg MB R3T 1B8
      </p>
      <p>
        Canada
      </p>
      <h3>
        shipping.method
      </h3>
      <p>
        Standard
      </p>
      <h3>
        payment.method
      </h3>
      <p>
        visa
      </p>
    </div>
    <div
      class="confirmation-help"
    >
      <p>
        confirmation.need_help
        <a
          href="#"
        >
          confirmation.contact_us
        </a>
      </p>
       
    </div>
    <div
      class="confirmation-continue"
    >
      <a
        class="confirmation-continue-button stx-button"
        href="/"
      >
        confirmation.continue_shopping
      </a>
    </div>
  </div>
</DocumentFragment>
`;
